agent-bootstrap/src/main/java/io/opentelemetry/auto/config/Config.java

```diff
@@ -401,7 +397,8 @@ private static String propertyNameToSystemPropertyName(final String setting) {
    * @throws NumberFormatException
    */
   private static <T> T valueOf(
-      final String value, @NonNull final Class<T> tClass, final T defaultValue) {
+      final String value, final Class<T> tClass, final T defaultValue) {
+    Objects.requireNonNull(tClass);
```

Comment:
This is ok, though also rare to validate parameters for a private method.

Reply:
I'll remove it, if dig into public methods which use this variable, they already have `Objects.requireNonNull` validation.